File: StructuralPattern/Bridge/Golang/Shape/shape.go
package shape

import (
	"fmt"
	c "v/color"
)

/**
 * Interface for shape
 */
type Shape interface {
	Draw() 
}

/**
 * Circle shape
 */
type Circle struct {
	color c.Color
}

// Constructor
func NewCircle(color c.Color) *Circle {
	return &Circle{color}
}

// Deloy draw 
func (circle *Circle) Draw() {
	fmt.Println("Circle: ", circle.color.GetColor())
}

/**
 * Square shape
 */
type Square struct {
	color c.Color
}

// Constructor
func NewSquare(color c.Color) *Square {
	return &Square{color}
}

// Deloy draw 
func (square *Square) Draw() {
	fmt.Println("Circle: ", square.color.GetColor())
}

File: StructuralPattern/Bridge/Python/Shape/Shape.py
"""
    Define shape
"""

from Color import Color

class Shape:
    """
        Interface for shape
    """
    def __init__(self, color):
        self.color = color
    
    def __str__(self):
        pass

class Circle(Shape):
    """
        Circle implements shape
    """
    def __init__(self,color):
        super().__init__(color)

    def __str__(self):
        return "Circle" + self.color.draw()
    
class Square(Shape):
    """
        Square implements shape
    """
    def __init__(self,color):
        super().__init__(color)

    def __str__(self):
        return "Square" + self.color.draw()
    